clash-prelude/src/Clash/Explicit/Mealy.hs

```diff
@@ -27,17 +28,56 @@ import           Clash.Explicit.Signal
   (KnownDomain, Bundle (..), Clock, Reset, Signal, Enable, register)
 import           Clash.XException      (NFDataX)
 
+import           Control.Monad.State.Strict
+  (State, runState)
+
 {- $setup
 >>> :set -XDataKinds -XTypeApplications
->>> import Clash.Explicit.Prelude
+>>> import Clash.Explicit.Prelude as C
 >>> import qualified Data.List as L
+>>> import Control.Lens (Lens', (%=), (-=), uses, use)
+>>> import Control.Monad.State.Strict (State)
 >>> :{
 let macT s (x,y) = (s',s)
       where
         s' = x * y + s
 :}
 
 >>> let mac clk rst en = mealy clk rst en macT 0
+>>> data DelayState = DelayState { _delayed :: Vec 4 Int , _untilValid :: Index 4 }
+>>> :{
+let delayed :: Lens' DelayState (Vec 4 Int)
+    delayed f = \(DelayState d u) -> (`DelayState` u) <$> f d
+    untilValid :: Lens' DelayState (Index 4)
+    untilValid f = \(DelayState d u) -> (DelayState d) <$> f u
+    initialDelayState = DelayState (C.repeat 0) maxBound
+:}
+>>> :{
+  let delayTop :: Clock System -> Reset System -> Enable System -> (Signal System Int -> Signal System (Maybe Int))
+      topDelay = P.undefined
+:}
+>>> let hello = "hello"
```

Comment:
Will be removed, currently trying to debug why the doctests are failing to compile the setup for `topDelay`, which should be defined as commented below.